lib/ASRouter.jsm

```diff
@@ -510,12 +511,19 @@ class _ASRouter {
     this.dispatch = this.dispatch.bind(this);
   }
 
+  _shouldShowExtendedTriplet(msg) {
+    return (
+      msg.template === EXTENDED_TRIPLETS_TEMPLATE &&
+      this.state.showExtendedTriplets
+    );
+  }
+
   async onPrefChange(prefName) {
     if (TARGETING_PREFERENCES.includes(prefName)) {
       // Notify all tabs of messages that have become invalid after pref change
       const invalidMessages = [];
       for (const msg of this._getUnblockedMessages()) {
-        if (!msg.targeting) {
+        if (!msg.targeting || this._shouldShowExtendedTriplet(msg)) {
           continue;
         }
         const isMatch = await ASRouterTargeting.isMatch(msg.targeting);
```

Comment:
Hm.. why is the extended triplets message no longer matching after a pref change? Its targeting doesn't explicitly check for logged in ??
https://github.com/mozilla/activity-stream/blob/2df1c58f59b34658d7eba6518b16ab6f51a2f11a/lib/OnboardingMessageProvider.jsm#L111-L112

Reply:
yeah why is `isMatch` failing?

Reply:
not sure about https://searchfox.org/mozilla-central/source/toolkit/components/utils/FilterExpressions.jsm#54 code here which is what isMatch uses?

Reply:
Looks like this `isMatch` is missing a 2nd argument for `context` ?

Reply:
`evaluateExpression` passes in a context while `findAllMessages` just uses a helper method:
https://github.com/mozilla/activity-stream/blob/2df1c58f59b34658d7eba6518b16ab6f51a2f11a/lib/ASRouter.jsm#L1132-L1142

Reply:
Noticed context used with FilterExpressions on PrefChange is missing 
```
messageImpressions: {…}
previousSessionEnd: 0
trailheadInterrupt: "join"
trailheadTriplet: "supercharge"
```

will try passing customContext using helper method and see if that helps
![context](https://user-images.githubusercontent.com/597734/64379082-00d51080-cfe3-11e9-8b0c-f81b6dee1642.png)